flow/aggregate.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) 2020 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+from collections.abc import Iterable
+from copy import deepcopy
+from itertools import groupby
+from itertools import zip_longest
+from functools import partial
+
+import signac
+from signac.contrib.project import Project, JobsCursor
+
+
+class aggregate:
+    """Decorator for operation functions that are to be aggregated.
+
+    If this class is used for aggregation then by-default, if the aggregator
+    parameter is not passed, an aggregate of all jobs will be created.
+
+    .. code-block:: python
+
+        example_aggregate = aggregate()
+        @example_aggregate
+        @FlowProject.operation
+        def foo(*jobs):
+            print(len(jobs))
+
+    :param aggregator:
+        Information on how to aggregate jobs. Takes in a list of
+        jobs and can return or yield lists or single jobs instead.
+        The default behavior is creating an aggregate of all jobs.
+    :type aggregator:
+        callable
+    :param sort:
+        Before aggregating, sort the jobs given by a statepoint parameter.
+        The default value is None.
+    :type sort:
+        str or NoneType
+    :param reverse:
+        States if the jobs are to be sorted in reverse order.
+        The default value is False.
+    :type reverse:
+        bool
+    :param select:
+        Condition for filtering jobs. This operates on a single job.
```

Comment:
```suggestion
        Condition for filtering jobs.
```

Why is it important to specify that this operates on a single job?

Reply:
Actually the point I was trying to make here is the sequence in which we currently go.
1. select
2. sort
3. aggregate

The `select` parameter should not be something like
```
def _select(job1, job2):
    return job1 > job2
```

Reply:
```suggestion
        Condition for filtering individual jobs. This is passed as the callable argument
        to `filter`.
```